### PR TITLE
Enhancement(Integration): Pinterest Tag event_id added for deduplication + custom event support

### PR DESCRIPTION
**Pull request by shrouti1507**
Adding deduplication support for pinterest_tag
Ref: https://view.highspot.com/viewer/629a4a9284742fae381c6804
Also, adding "custom" event support, if event sent by the user doesn't match the internal mapping or event mapping in webapp.

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

- - -
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/635)

### CODE REVIEW

File: integrations/PinterestTag/browser.js [hunk 1]
```diff
@@ -211,6 +217,7 @@ export default class PinterestTag {
       destEvent?.hasEmptyProducts,
       destEvent?.isUserDefinedEvent
     );
+    pinterestObject.event_id = this.deduplicationKey || messageId;
```
Comment (utsabc) [inline integrations/PinterestTag/browser.js:220]: Shouldn't this be
`get(message,this.deduplicationKey)`

File: integrations/PinterestTag/browser.js [hunk 2]
```diff
@@ -193,14 +191,22 @@ export default class PinterestTag {
         };
       }
     }
+    const destinationEvent = eventMapping.find((p) =>
+      p.src.includes(event.toLowerCase())
+    );
+    if (!destinationEvent) {
+      return {
+        dest: "custom",
+      };
```
Comment (utsabc) [inline integrations/PinterestTag/browser.js:200]: Does Pinterest support custom?

Comment (shrouti1507) [reply integrations/PinterestTag/browser.js:200]: I can see custom call in their dashboard.
Screenshot 2022-09-01 at 11 58 23 AM

Comment (utsabc) [reply integrations/PinterestTag/browser.js:200]: ok

File: integrations/PinterestTag/browser.js (hunk 1 above)
Comment (shrouti1507) [reply integrations/PinterestTag/browser.js]: corrected.

**Review by utsabc — DISMISSED** [review]

**Review by MoumitaM — APPROVED** [review]

**Review by utsabc — APPROVED** [review]

**Pull request merged**